adapters/handlers/rest/middlewares.go

```diff
@@ -219,8 +221,10 @@ func addLiveAndReadyness(state *state.State, next http.Handler) http.Handler {
 
 		if r.URL.String() == "/v1/.well-known/ready" {
 			code := http.StatusServiceUnavailable
-			if state.ClusterService.Ready() && state.Cluster.ClusterHealthScore() == 0 {
-				code = http.StatusOK
+			if !entcfg.Enabled(os.Getenv("MAINTENANCE_MODE")) {
+				if state.ClusterService.Ready() && state.Cluster.ClusterHealthScore() == 0 {
+					code = http.StatusOK
+				}
```

Comment:
syntax is off, but something like this would enable us to set the environment variable on the whole cluster instead of needing to modify nodes individually: eg `MAINTENANCE_NODES=weaviate-3`:

```suggestion
                         localHostname := state.Cluster.LocalName()
			maintenanceNodes := os.Getenv("MAINTENANCE_NODES")
			if slices.Index(strings.Split(maintenanceNodes, ","), localHostname) == -1 {
				if state.ClusterService.Ready() && state.Cluster.ClusterHealthScore() == 0 {
					code = http.StatusOK
				}
			}
```

Reply:
this is overkill for now